File: app/models/review.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.sql import func
from app.db.database import Base


class Review(Base):
    __tablename__ = "reviews"

    id = Column(Integer, primary_key=True, index=True)
    movie_id = Column(Integer, index=True, nullable=False)
    user_id = Column(Integer, index=True, nullable=False)
    rating = Column(Integer, nullable=False)
    title = Column(String, nullable=True)
    body = Column(String, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

File: intelligent-movie-frontend/src/app/models/review.ts
export interface ReviewCreate {
  rating: number;
  title?: string;
  body: string;
}

export interface ReviewUpdate {
  rating?: number;
  title?: string;
  body?: string;
}

export interface ReviewRead {
  id: number;
  movie_id: number;
  user_id: number;
  rating: number;
  title?: string;
  body: string;
  created_at?: string;
  updated_at?: string;
  likes?: number;
  dislikes?: number;
}
export interface Review {
}
